### PR TITLE
opensnitch: 1.6.4 -> 1.6.5, opensnitch-ui: 1.6.4 -> 1.6.5

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

- update opensnitch and opensnitch-ui to 1.6.5
- make rules path `Rules.Path` configurable
- drop DefaultDuration option, it's deprecated
- make `Ebpf.ModulesPath configurable
- dynamic config file path
- remove static paths to ebpf modules
- move opensnitch packages to pkgs/by-name
- added meta maintainer to the opensnitch module
- move creation of system-fw.json to systemd.tmpfiles

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by onny** [comment]
dont know how to fix this error yet

```
       error: attribute 'services' missing

       at /nix/store/g98g0sg6w7jcak14cckfwdc9hnq5ppj7-nixos/modules/services/security/opensnitch.nix:6:9:

            5| let
            6|   cfg = config.services.opensnitch;
             |         ^
            7|   format = pkgs.formats.json {};
Cacheable portion of option doc build failed.
Usually this means that an option attribute that ends up in documentation (eg `default` or `description`) depends on the restricted module arguments `config` or `pkgs`.

Rebuild your configuration with `--show-trace` to find the offending location. Remove the references to restricted arguments (eg by escaping their antiquotations or adding a `defaultText`) or disable the sandboxed build for the failing module by setting `meta.buildDocsInSandbox = false`.

error: builder for '/nix/store/n9qmydpszxdn9x1fxbqg26n6px19p6hf-lazy-options.json.drv' failed with exit code 1;
       last 10 log lines:
       >
       >             5| let
       >             6|   cfg = config.services.opensnitch;
       >              |         ^
       >             7|   format = pkgs.formats.json {};
       > Cacheable portion of option doc build failed.
       > Usually this means that an option attribute that ends up in documentation (eg `default` or `description`) depends on the restricted module arguments `config` or `pkgs`.
       >
       > Rebuild your configuration with `--show-trace` to find the offending location. Remove the references to restricted arguments (eg by escaping their antiquotations or adding a `defaultText`) or disable the sandboxed build for the failing module by setting `meta.buildDocsInSandbox = false`.
       > 
       For full logs, run 'nix log /nix/store/n9qmydpszxdn9x1fxbqg26n6px19p6hf-lazy-options.json.drv'.
error: 1 dependencies of derivation '/nix/store/y6b23j3hkh1w2m7nhw5l43xqym0zs4vm-options.json.drv' failed to build
error: 1 dependencies of derivation '/nix/store/g34z9z09xb5vxzh2ib90npji76zc9hrp-nixos-manual-html.drv' failed to build
Error: Process completed with exit code 1.
```

**Comment by onny** [comment]
Does someone know how to fix this? @wegank @happysalada ?

**Comment by wegank** [comment]
Try setting `meta.buildDocsInSandbox = false` in `nixos/modules/services/security/opensnitch.nix`?

**Comment by onny** [comment]
Thank you @wegank, was able to fix it :)

File: pkgs/by-name/op/opensnitch/go.mod [hunk 1]
```diff
@@ -1,19 +1,19 @@
 module github.com/evilsocket/opensnitch/daemon
```
Comment (SuperSandro2000) [inline pkgs/by-name/op/opensnitch/go.mod:1]: Why do we need to vendor this file? it exists uostream https://github.com/evilsocket/opensnitch/blob/master/daemon/go.mod

File: nixos/modules/services/security/opensnitch.nix [hunk 2]
```diff
@@ -151,40 +167,46 @@ in {
...
+            rules} \) \
+          ''} -delete
+          ${concatMapStrings ({ file, local }: ''
+            ln -sf '${file}' "${local}"
+          '') rules}
+
+          if [ ! -f /etc/opensnitchd/system-fw.json ]; then
+            mkdir -p /etc/opensnitchd
+            cp "${pkgs.opensnitch}/etc/opensnitchd/system-fw.json" "/etc/opensnitchd/system-fw.json"
+          fi
```
Comment (SuperSandro2000) [inline nixos/modules/services/security/opensnitch.nix:200]: Can this be done via tmpfiles?

File: pkgs/by-name/op/opensnitch/go.mod (hunk 1 above)
Comment (onny) [reply pkgs/by-name/op/opensnitch/go.mod:1]: Without it I get this error:
```
       > go: downloading github.com/BurntSushi/toml v0.4.1
       > go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
       > go: downloading golang.org/x/mod v0.5.1
       > go: github.com/evilsocket/opensnitch/daemon/firewall/nftables/nftest: package github.com/evilsocket/opensnitch/daemon/firewall/nftables/nftest imports github.com/vishvananda/netns from implicitly required module; to add missing requirements, run:
       >       go get github.com/vishvananda/netns@v0.0.0-20200728191858-db3c7e526aae
```

File: nixos/modules/services/security/opensnitch.nix (hunk 2 above)
Comment (onny) [reply nixos/modules/services/security/opensnitch.nix:200]: good point 👍 thanks for the hint

**Pull request merged**